DEHPMatlab/ViewModel/Dialogs/DstConnectViewModel.cs

```diff
@@ -0,0 +1,172 @@
+// 
+// The DEHPMatlab is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace DEHPMatlab.ViewModel.Dialogs
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reactive;
+    using System.Threading.Tasks;
+
+    using DEHPCommon.UserInterfaces.Behaviors;
+    using DEHPCommon.UserInterfaces.ViewModels.Interfaces;
+
+    using DEHPMatlab.DstController;
+    using DEHPMatlab.ViewModel.Dialogs.Interfaces;
+    using DEHPMatlab.Views.Dialogs;
+
+    using ReactiveUI;
+
+    /// <summary>
+    /// The view model for <see cref="DstConnect"/>
+    /// </summary>
+    public class DstConnectViewModel : ReactiveObject, IDstConnectViewModel, ICloseWindowViewModel
+    {
+        /// <summary>
+        /// Contains the base of all Matlab ProgId
+        /// </summary>
+        private const string BaseMatlabKey = "Matlab.Application";
+
+        /// <summary>
+        /// Backing Field for <see cref="IsBusy"/>
+        /// </summary>
+        private bool isBusy;
+
+        /// <summary>
+        /// Backing field for <see cref="SelectedMatlabVersion"/>
+        /// </summary>
+        private KeyValuePair<string, string> selectedMatlabVersion;
+
+        /// <summary>
+        /// Backing field for <see cref="ErrorMessageText"/>
+        /// </summary>
+        private string errorMessageText;
+
+        /// <summary>
+        /// The <see cref="IDstController"/>
+        /// </summary>
+        private readonly IDstController dstController;
+
+        /// <summary>
+        /// Initialize a new instance of <see cref="DstConnectViewModel"/>
+        /// </summary>
+        /// <param name="dstController">The <see cref="IDstController"/></param>
+        public DstConnectViewModel(IDstController dstController)
+        {
+            this.dstController = dstController;
+            this.MatlabVersionDictionary = new Dictionary<string, string>();
+            this.PopulateDictionary();
+            this.ConnectCommand = ReactiveCommand.CreateAsyncTask(async _ => await this.ConnectCommandExecute());
+        }
+
+        /// <summary>
+        /// Populate the <see cref="MatlabVersionDictionary"/>
+        /// </summary>
+        private void PopulateDictionary()
+        {
+            this.MatlabVersionDictionary["latest"] = "Lastest Installed Version";
+            this.MatlabVersionDictionary["9.11"] = "Matlab R2021b";
```

Comment:
This can be done a bit smarter in terms of repetition and being prone to errors.